### PR TITLE
PGMT-25952

### PR DESCRIPTION
**Pull request by ghost**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

File: modules/zetaBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import {BANNER} from "../src/mediaTypes";
+const BIDDER_CODE = 'Zeta Global';
+const ENDPOINT_URL = 'http://prebid-us-east.rfihub.com:2080/prebid';
```
Comment (jkushmaul) [inline modules/zetaBidAdapter.js]: this should be a geo routed GTM property, will exclude Geo in the name.

File: modules/zetaBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+  /**
+     * Make a server request from the list of BidRequests.
+     *
+     * @param {validBidRequests[]} - an array of bidRequest objects
+     * @param {bidderRequest} - master bidRequest object
+     * @return ServerRequest Info describing the request to the server.
+     */
+  buildRequests: function(validBidRequests, bidderRequest) {
+    const secure = location.protocol.indexOf('https') > -1 ? 1 : 0;
```
Comment (jkushmaul) [inline modules/zetaBidAdapter.js:30]: We should only support secure ads.

File: modules/zetaBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    const params = validBidRequests[0].params;
+    let data = {
+      id: validBidRequests[0].bidId,
+      secure,
+      banner: buildBanner(validBidRequests[0])
+    };
+    let isMobile = /(ios|ipod|ipad|iphone|android)/i.test(navigator.userAgent) ? 1 : 0;
```
Comment (jkushmaul) [inline modules/zetaBidAdapter.js:37]: you can add dalvik and darwin to catch in-app useragents which may not have "android" or "ios" etc within.

File: integrationExamples/gpt/hello_world.html [hunk 4]
```diff
@@ -24,9 +24,14 @@
...
-                bidder: 'appnexus',
+                bidder: 'Zeta Global',
                 params: {
-                    placementId: 13144370
+                    placementId: 13144370,
+                    user: {
+                        buyeruid: "1992631735303440485",
+                        id: "697565373385267402"
+                    },
+                    ip: "0.0.0.0"
```
Comment (jkushmaul) [inline integrationExamples/gpt/hello_world.html:34]: Client side js, right? How is this being determined from js? It's not generally available until the request is received on the server.  Unless you fetch it from a service that returns that as a response. It's probably already available as the request ip on the adapter.

File: modules/zetaBidAdapter.js (hunk 1 above)
Comment (jkushmaul) [reply modules/zetaBidAdapter.js]: also if this is going to be a browser endpoint, this needs to be https://prebid.rfiub.com/ (port 443)

File: integrationExamples/gpt/hello_world.html [hunk 5]
```diff
@@ -24,9 +24,14 @@
             },
             // Replace this object to test a new Adapter!
             bids: [{
-                bidder: 'appnexus',
+                bidder: 'Zeta Global',
```
Comment (bt-wil) [inline integrationExamples/gpt/hello_world.html:27]: How does the replacement work for individual companies?  Wouldn't the next company overwrite these values that we've put in?

File: integrationExamples/gpt/hello_world.html [hunk 6]
```diff
@@ -24,9 +24,14 @@
...
                 params: {
-                    placementId: 13144370
+                    placementId: 13144370,
+                    user: {
+                        buyeruid: "1992631735303440485",
```
Comment (bt-wil) [inline integrationExamples/gpt/hello_world.html:31]: Are there any linters in the project?  single quote vs double quotes.
Also, are these specific to a real/test user in our database or are these just random ids?

File: modules/zetaBidAdapter.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+  /**
+     * Make a server request from the list of BidRequests.
+     *
+     * @param {validBidRequests[]} - an array of bidRequest objects
+     * @param {bidderRequest} - master bidRequest object
+     * @return ServerRequest Info describing the request to the server.
+     */
+  buildRequests: function(validBidRequests, bidderRequest) {
+    const secure = location.protocol.indexOf('https') > -1 ? 1 : 0;
+    const params = validBidRequests[0].params;
```
Comment (bt-wil) [inline modules/zetaBidAdapter.js:31]: Is only the first request used?  Let's extract this to a variable if that's the case.

File: modules/zetaBidAdapter.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    let data = {
+      id: validBidRequests[0].bidId,
+      secure,
```
Comment (bt-wil) [inline modules/zetaBidAdapter.js:34]: the data is a dict? what is the `secure` key?

File: modules/zetaBidAdapter.js (hunk 3 above)
Comment (bt-wil) [reply modules/zetaBidAdapter.js:37]: does the prebid have mobile checks or utilize some other lib for the check?  i'm not sure how accurate this is, and if we need to (remember to) updated this whenever a "new" type of mobile devices comes out.

File: modules/zetaBidAdapter.js [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+     */
+  interpretResponse: function(serverResponse, bidRequest) {
+    const ttl = 200;
+    const netRev = true;
+    let bidResponse = [];
+    let bids = serverResponse.body.seatbid || [];
+    let cur = serverResponse.body.cur;
+    let bid = {
+      requestId: serverResponse.body.id,
+      cpm: bids[0].bid[0].price,
```
Comment (bt-wil) [inline modules/zetaBidAdapter.js:77]: error checks?  and let's extract to variable.

File: modules/zetaBidAdapter.js [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+
+  /**
+     * Register the user sync pixels which should be dropped after the auction.
+     *
+     * @param {SyncOptions} syncOptions Which user syncs are allowed?
+     * @param {ServerResponse[]} serverResponses List of server's responses.
+     * @return {UserSync[]} The user syncs which should be dropped.
+     */
+  getUserSyncs: function(syncOptions, serverResponses, gdprConsent, uspConsent) {
+    const syncs = []
```
Comment (bt-wil) [inline modules/zetaBidAdapter.js:98]: is semicolon needed?

File: modules/zetaBidAdapter.js [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+      if (syncOptions.iframeEnabled) {
```
Comment (bt-wil) [inline modules/zetaBidAdapter.js:99]: indentation.

File: modules/zetaBidAdapter.js [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+          url: USER_SYNC_URL
+        });
+      }
+    return syncs;
+  }
+}
+
+function buildBanner(bid) {
+  let sizes = [];
+  bid.mediaTypes && bid.mediaTypes.banner && bid.mediaTypes.banner.sizes ? sizes = bid.mediaTypes.banner.sizes : sizes = bid.sizes;
```
Comment (bt-wil) [inline modules/zetaBidAdapter.js:111]: if/else might be more readable.

File: modules/zetaBidAdapter.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+        {
+            code: 'test-div',
+            mediaTypes: {
+                banner: {
+                    sizes: [[300, 250]],  // a display size
+                }
+            },
+            bids: [
+                {
+                    bidder: "example",
```
Comment (bt-wil) [inline modules/zetaBidAdapter.md:25]: consistent on the quote

File: modules/zetaBidAdapter.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+                    params: {
+                        placement: '12345'
```
Comment (bt-wil) [inline modules/zetaBidAdapter.md:27]: string or int?

File: modules/zetaBidAdapter.js [hunk 15]
```diff
@@ -0,0 +1,118 @@
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import {BANNER} from "../src/mediaTypes";
+const BIDDER_CODE = 'Zeta Global';
+const ENDPOINT_URL = 'https://prebid.rfihub.com/prebid';
+const USER_SYNC_URL = 'http://p.rfihub.com/cm?pub=42770&in=1';
```
Comment (bt-wil) [inline modules/zetaBidAdapter.js:5]: secure url available for this endpoint?

**Pull request closed without merging**